Neos.Neos/Classes/Utility/LegacyNodePathNormalizer.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Utility;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\AbsoluteNodePath;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindClosestNodeFilter;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\Projection\ContentGraph\NodePath;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Flow\Annotations as Flow;
+use Neos\Neos\Domain\Service\NodeTypeNameFactory;
+
+/**
+ * @deprecated the syntax is deprecated and not part of the content repository core
+ * @internal implementation detail of Neos.Neos:NodeUri
+ */
+final class LegacyNodePathNormalizer
+{
+    #[Flow\Inject]
+    protected ContentRepositoryRegistry $contentRepositoryRegistry;
+
+    /**
+     * Converts legacy paths to the corresponding AbsoluteNodePath depending on the passed base node.
+     *
+     * Supported legacy syntax:
+     *
+     *  - /sites/site/absolute/path
+     *  - ~/site-relative/path
+     *  - ~
+     *
+     * The following syntax is not implemented and handled here:
+     *
+     *  - /<Neos.Neos:Sites>/my-site/main
+     *  - some/relative/path
+     *
+     * Also while legacy and previously allowed, node path traversal like ./neos/info or ../foo/../../bar is not handled.
+     */
+    public function tryResolveLegacyPathSyntaxToAbsoluteNodePath(
+        string $path,
+        Node $baseNode
+    ): ?AbsoluteNodePath {
```

Comment:
I don't understand the return type. This can either return an `AbsoluteNodePath` or lead to an exception _or_ it returns `null`. What is the meaning of `null` and is that a good abstraction? Maybe we can at least add some `@return` doc comment explaining the ambiguity?

Reply:
Ah I see, in the usages you do
```
tryResolveLegacyPathSyntaxToAbsoluteNodePath($path, ..) ?? $path
```
wouldn't it make things easier to have some
```php
$absoluteNodePath = AbsoluteNodePath::tryFromString($path);
if ($absoluteNodePath !== null) {
    return $absoluteNodePath;
}
```
at the beginning of this method and get rid of the nullable return type?

Reply:
youre right to point this out:) I forgot to write down my line of thought here.

The idea is to separate the legacy behaviour from the more future proof, so we can later just remove the dependency to `tryResolveLegacyPathSyntaxToAbsoluteNodePath` ... calling `AbsoluteNodePath::tryFromString` would not be enough as the relative base node behaviour would also need to be handled ... 

its a little odd but its all `@internal` so maybe keep it this way?

Alternatively this could be the api:

```php
public function preprocessLegacyPathSyntax(
        string $path,
        Node $baseNode
    ): string;
```

Reply:
its a little weird but for now well keep it okay its internal :)

Reply:
thank you for the review though ❤️ though now its time to tag the beta:)

Reply:
I'm absolutely fine with being overruled, but it's a bit demotivating if all my comments are just disregarded. You could have at least added the `@return` doc comment.
Anyways, not a big deal in this case of course

Reply:
you are right @bwaidelich we had a short call about it, and decided this is ultimately throw away code and we shouldn't bikeshed about it and instead get the beta out. The doc comment would definitely not hurt though. 
SORRY! ❤️wasn't meant to ignore you just to get the topic off of our (collective) heads, we discussed the other options but didn't see any single one that was much more pretty and therefore worth changing to.

Reply:
no worries, no need to be sorry. But thanks for the explanation!